addons/my-ocular/discuss.js

```diff
@@ -1,41 +1,94 @@
 
-    if (userStatus) {
-      let br = document.createElement("br");
-      addon.tab.displayNoneWhileDisabled(br);
-      let status = document.createElement("i");
-      addon.tab.displayNoneWhileDisabled(status);
-      status.title = msg("status-hover");
-      status.innerText = userStatus;
+    let br = document.createElement("br");
+    addon.tab.displayNoneWhileDisabled(br);
 
-      let dot = document.createElement("span");
-      addon.tab.displayNoneWhileDisabled(dot, { display: "inline-block" });
-      dot.title = msg("status-hover");
-      dot.className = "my-ocular-dot";
+    // Create status element
+    let status = document.createElement("i");
+    status.classList.add("my-ocular-status");
+    addon.tab.displayNoneWhileDisabled(status);
+    updateTitle(status, statusType);
 
-      dot.style.backgroundColor = color;
+    // Create ocular dot
+    let dot = document.createElement("span");
+    addon.tab.displayNoneWhileDisabled(dot, { display: "inline-block" });
+    updateTitle(dot, statusType);
+    dot.className = "my-ocular-dot";
 
-      left.appendChild(br);
-      left.appendChild(status);
-      if (color) left.appendChild(dot);
+    left.appendChild(br);
+    left.appendChild(status);
+    left.appendChild(dot);
+
+    if (userStatus) {
+      status.innerText = userStatus;
+      dot.style.backgroundColor = statusType == "ocular" ? (color ? color : "#bbb") : "";
     }
   });
 
   async function fetchStatus(username) {
-    const response = await fetch(`https://my-ocular.jeffalo.net/api/user/${username}`);
-    const data = await response.json();
+    let statusType; // Why? Different posts can have different statuses
+    let response = await fetchSpecificStatus(username, addon.settings.get("show-status"));
+    // If status is not set, try to get another status
+    if (!response.status) {
+      let otherStatus =
+        addon.settings.get("show-status") == "aviate"
+          ? "ocular"
+          : addon.settings.get("show-status") == "ocular"
+          ? "aviate"
+          : "";
+      response = await fetchSpecificStatus(username, otherStatus);
+      if (response.status) {
+        statusSetting = statusType;
+        statusType = otherStatus;
+      }
+    } else if (response != false) {
+      statusType = addon.settings.get("show-status");
+    }
     return {
-      userStatus: data.status?.replace(/\n/g, " "),
-      color: data.color,
+      userStatus: response.status?.replace(/\n/g, " "),
+      color: response.color,
+      statusType: statusType,
     };
   }
+
+  async function fetchSpecificStatus(username, type) {
+    if (type == "ocular") {
+      return (await fetch(`https://my-ocular.jeffalo.net/api/user/${username}`)).json();
+    } else if (type == "aviate") {
+      if (await addon.auth.fetchIsLoggedIn())
+        return await (
+          await fetch(`https://aviateapp.eu.org/api/${username}?visitor=${await addon.auth.fetchUsername()}`)
```

Comment:
Sending the visitor over is a bit shady, don't you think?

Reply:
It's only used for the `{visitor}` component which returns the value of `?visitor=`.

Reply:
@Norbiros Please remove the visitor query param

Reply:
I understand. I wanted to parse it in client side, but MystiPi doesn't agree to add any client side code.

Reply:
I reverted changes.

Reply:
I just removed the component since it does seem kind of shady TBH. Sorry for the inconvenience.